### PR TITLE
DF-210: Make filters list in long filter page searchable

### PR DESCRIPTION
**Pull request by AshTNA**
Hi @JamesWChan 

Would you be able to merge this PR?

It relates to https://national-archives.atlassian.net/browse/DF-210

It adds a search component to the 'long filters' page on Search, which allows the list of filters to be narrowed down via a keyword. It works by:

- Grabbing the NodeList of filter`<li>` elements
- Converting the list into an Array
- Running `filter()` on this array to grab the narrowed down filters
- Rendering these narrowed down filters in the DOM

It will also always show any filters that the user has checked, so that they don't have to delete their keyword and then scroll through all the filters to see which ones are checked.

Accessibility wise, initially we were going to use `aria-live='polite'` to always update users on how many filters were being shown. However it was conflicting with the users' keyword being read out. The screen reader would try to read out how many filters were showing, but then it would cut out and read out the keyword the user had typed instead. Therefore, we are using `ariadescribed-by` on the fieldset to describe how many filters are remaining.

Access the component here: http://localhost:8000/search/catalogue/long-filter-chooser/collection/?q=&per_page=20&sort_order=asc&group=tna

It's been checked on Chrome, Firefox, IE11 and on mobile.

Thanks :+1:

image

### CODE REVIEW

**Comment by AshTNA** [comment]
I need to get the editorial checked over, so if you approve the changes please don't merge it yet

File: sass/includes/search/_long-filters.scss
Comment (JamesWChan) [inline sass/includes/search/_long-filters.scss:31]: Could this use the `screen__lg` variable? Or does it need to be 60rem (`screen__lg` is 62rem)?

File: sass/includes/search/_long-filters.scss
Comment (JamesWChan) [inline sass/includes/search/_long-filters.scss:59]: Would we want to change this from `px` to `rem`?

File: sass/includes/search/_long-filters.scss
Comment (JamesWChan) [inline sass/includes/search/_long-filters.scss:97]: Could we use the `screen__lg` variable here?

File: sass/includes/search/_long-filters.scss
Comment (JamesWChan) [inline sass/includes/search/_long-filters.scss:108]: Would we want to change this from px to rem?

File: scripts/src/modules/search/search-long-filters.js
Comment (JamesWChan) [inline scripts/src/modules/search/search-long-filters.js:81]: Quick question - you've declared this variable on line 45; does it need redeclaring here or could it just be re-assigned?

File: scripts/src/modules/search/search-long-filters.js
Comment (JamesWChan) [inline scripts/src/modules/search/search-long-filters.js:53]: Could this be simplified to:

`if(e.key === “Enter”) { 
      e.preventDefault(); 
  }`
  
  And then replace line 94 with something like `$searchBox.addEventListener('keypress', handleSearch);`?
  
Then I think `disableSearchSubmit` wouldn't be needed. Just my opinion though so if it's preferable to keep it as it is then that's cool with me.

**Comment by JamesWChan** [comment]
Hi @AshTNA,

Just left a couple of comments, everything is working well so they're more questions than anything. Also I spotted a small issue on IE11 and Safari with the search box display, screenshot below:

image

File: scripts/src/modules/search/search-long-filters.js
Comment (JamesWChan) [reply scripts/src/modules/search/search-long-filters.js:53]: Actually, this might not be a good idea. Just realised that this would result in calling the `handleSearch` function twice every time a key is pressed.

Comment (AshTNA) [reply scripts/src/modules/search/search-long-filters.js:53]: Thanks @JamesWChan  - I also think the `preventDefault` has to be on the `$parentForm` element, as with forms it will consider any Enter keypress while inside the form as a submit, so it can't be on the `$searchBox` itself. The `id` checking is so that the submit buttons and other form elements can still trigger a submission by pressing enter.

**Comment by AshTNA** [comment]
Thanks @JamesWChan - everything should be fixed. Feel free to approve then I'll get the editorial checked before merging :+1:

**Review by JamesWChan — APPROVED** [review]
All good 👍

**Pull request merged**
